src/auth/auth.cpp

```diff
@@ -569,18 +565,40 @@ void Auth::DeleteDatabase(const std::string &db, system::Transaction *system_tx)
       SaveUser(*user, system_tx);
     }
   }
+  for (auto it = storage_.begin(kRolePrefix); it != storage_.end(kRolePrefix); ++it) {
+    auto rolename = it->first.substr(kRolePrefix.size());
+    if (auto role = GetRole(rolename)) {
+      role->db_access().Delete(db);
+      SaveRole(*role, system_tx);
+    }
+  }
 }
 
 bool Auth::SetMainDatabase(std::string_view db, const std::string &name, system::Transaction *system_tx) {
   if (auto user = GetUser(name)) {
-    if (!user->db_access().SetDefault(db)) {
-      throw AuthException("Couldn't set default database '{}' for user '{}'!", db, name);
-    }
-    SaveUser(*user, system_tx);
+    SetMainDatabase(db, *user, system_tx);
```

Comment:
TODO rename to Default (not Main)

Reply:
Renamed `Auth::Database::SetDefault` to `SetMain`
SET MAIN DATABASE is the query so I'll reflect that in the code